miniapp/miniapp_eigensolver.cpp

```diff
@@ -0,0 +1,173 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2022, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#include <iostream>
+#include <limits>
+
+#include <pika/init.hpp>
+#include <pika/program_options.hpp>
+#include <pika/runtime.hpp>
+
+#include "dlaf/common/format_short.h"
+#include "dlaf/common/index2d.h"
+#include "dlaf/common/range2d.h"
+#include "dlaf/common/timer.h"
+#include "dlaf/communication/communicator_grid.h"
+#include "dlaf/communication/init.h"
+#include "dlaf/eigensolver/eigensolver.h"
+#include "dlaf/init.h"
+#include "dlaf/matrix/copy.h"
+#include "dlaf/matrix/matrix.h"
+#include "dlaf/matrix/matrix_mirror.h"
+#include "dlaf/miniapp/dispatch.h"
+#include "dlaf/miniapp/options.h"
+#include "dlaf/types.h"
+
+namespace {
+using dlaf::Device;
+using dlaf::Backend;
+using dlaf::DefaultDevice_v;
+using dlaf::SizeType;
+using dlaf::GlobalElementSize;
+using dlaf::TileElementSize;
+using dlaf::Matrix;
+using dlaf::matrix::MatrixMirror;
+using dlaf::common::Ordering;
+using dlaf::comm::Communicator;
+using dlaf::comm::CommunicatorGrid;
+
+struct Options
+    : dlaf::miniapp::MiniappOptions<dlaf::miniapp::SupportReal::Yes, dlaf::miniapp::SupportComplex::Yes> {
+  SizeType m;
+  SizeType mb;
+  blas::Uplo uplo;
+
+  Options(const pika::program_options::variables_map& vm)
+      : MiniappOptions(vm), m(vm["matrix-size"].as<SizeType>()), mb(vm["block-size"].as<SizeType>()),
+        uplo(dlaf::miniapp::parseUplo(vm["uplo"].as<std::string>())) {
+    DLAF_ASSERT(m > 0, m);
+    DLAF_ASSERT(mb > 0, mb);
+
+    if (do_check != dlaf::miniapp::CheckIterFreq::None) {
+      std::cerr << "Warning! At the moment result checking it is not implemented." << std::endl;
+      do_check = dlaf::miniapp::CheckIterFreq::None;
+    }
```

Comment:
Just one comment/doubt.

For `--check-result` there is both the check in the `Options` c'tor and the `DLAF_UNIMPLEMENTED`. For "local" check there is just this unimplemented. Wouldn't be better to add a check in `Options` constructor too with the hint about how to setup the flags?

Something like this in 

```
DLAF_ASSERT((vm["grid-rows"].as<int>() * vm["grid-cols"].as<int>()) == 1,
              "Error! Distributed eigensolver is not avilable yet. "
              "Please rerun with both --grid-rows and --grid-cols set to 1");
```